warehouse/malware/checks/pkg_info_patterns/check.py

```diff
@@ -0,0 +1,115 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from textwrap import dedent
+
+import yara
+
+from warehouse.malware.checks.base import MalwareCheckBase
+from warehouse.malware.checks.utils import extract_file_content,\
+    fetch_url_content, extract_file_list
+from warehouse.malware.errors import FatalCheckError
+from warehouse.malware.models import VerdictClassification, VerdictConfidence
+
+
+class PkgInfoPatternCheck(MalwareCheckBase):
+    _yara_rule_file = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "pkg_info_rules.yara"
+    )
+
+    version = 3
+    short_description = "A check for common malicious patterns in PKG-INFO"
+    long_description = dedent(
+        """
+        This check uses YARA to search for common malicious patterns in the PKG-INFO
+        files of uploaded release archives.
+        """
+    )
+    check_type = "event_hook"
+    hooked_object = "File"
+
+    def __init__(self, db):
+        super().__init__(db)
+        self._yara_rules = self._load_yara_rules()
+
+    def _load_yara_rules(self):
+        return yara.compile(filepath=self._yara_rule_file)
+
+    def scan(self, **kwargs):
+        release_file = kwargs.get("obj")
+        file_url = kwargs.get("file_url")
+        if release_file is None or file_url is None:
+            raise FatalCheckError(
+                "Release file or file url is None, indicating user error."
+            )
+
+        archive_stream = fetch_url_content(file_url)
+
+        file_list = extract_file_list(archive_stream)
```

Comment:
I'm a bit wary of extracting user-submitted files here, as we don't have any sort of sandboxing in place.

Reply:
https://github.com/pypa/warehouse/blob/main/warehouse/malware/checks/setup_patterns/check.py#L63

But one of existing modules already does it)